src/test/java/com/networknt/schema/BaseJsonSchemaValidatorTest.java

```diff
@@ -49,7 +49,7 @@ protected JsonNode getJsonNodeFromUrl(String url) throws Exception {
     }
 
     protected JsonSchema getJsonSchemaFromClasspath(String name) throws Exception {
-        JsonSchemaFactory factory = JsonSchemaFactory.getInstance();
+        JsonSchemaFactory factory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V4);
```

Comment:
I'm away from my IDE, so I don't know what the subclasses are of this class. Anyway, wouldn't it be better to allow the SpecVersion to be passed in through a protected constructor? That would allow the subclasses to easily choose the spec version they need to use. The other option is to make the SpecVersion an additional argument to these methods.

Reply:
The JsonShemaFactory is a singleton at the moment. It is designed this way to prevent multiple instances to be created. I am open to making it more flexible for users.

Reply:
I get that it's a singleton. I was talking about subclasses of the BaseJsonSchemaValidatorTest. I was meaning something like the following, so that dependency injection is possible:

```
class BaseJsonSchemaValidatorTest {
  private final SpecVersion.VersionFlag version;
  protected BaseJsonSchemaValidatorTest(SpecVersion.VersionFlag version) {
    this.version = version
  }

  protected JsonSchema getJsonSchemaFromClasspath(String name) throws Exception {
    JsonSchemaFactory factory = JsonSchemaFactory.getInstance(this.version);
    ...
  }
}
```

Reply:
Yes. It works much better and flexible.